src/edu/stanford/nlp/sempre/interactive/DALAnalyzer.java

```diff
@@ -73,49 +75,95 @@ else if (c == '>' || c == '<') {
       } else if (c == '=') {
         buf.append(c);
         buf.append(' ');
-      } else
+      }
+      else
         buf.append(c);
       if (separate)
         buf.append(' ');
     }
     utterance = buf.toString().trim();
     if (!utterance.equals("")) {
       String[] tokens = utterance.split("\\s+");
-      for (String token : tokens) {
-        String lemma = token;
-        if (token.endsWith("s") && token.length() > 1)
-          lemma = token.substring(0, token.length() - 1);
-
-        languageInfo.tokens.add(LanguageAnalyzer.opts.lowerCaseTokens ? token.toLowerCase() : token);
-        languageInfo.lemmaTokens.add(LanguageAnalyzer.opts.lowerCaseTokens ? lemma.toLowerCase() : lemma);
-
-        // Is it a written out number?
-        int x = Arrays.asList(numbers).indexOf(token);
-        if (x != -1) {
-          languageInfo.posTags.add("CD");
-          languageInfo.nerTags.add("NUMBER");
-          languageInfo.nerValues.add(x + "");
-          continue;
-        }
-
-        try {
-          Double.parseDouble(token);
-          languageInfo.posTags.add("CD");
-          languageInfo.nerTags.add("NUMBER");
-          languageInfo.nerValues.add(token);
-        } catch (NumberFormatException e) {
-          // Guess that capitalized nouns are proper
-          if (Character.isUpperCase(token.charAt(0)))
-            languageInfo.posTags.add("NNP");
-          else if (token.equals("'") || token.equals("\"") || token.equals("''") || token.equals("``"))
-            languageInfo.posTags.add("''");
-          else
+      int parenCount=0;//+1 for ( -1 for )
```

Comment:
do this as a mode during tokenization, not post processing, that should make life much easier.